fog/ledger/server/tests/router_integration.rs

```diff
@@ -106,40 +106,44 @@ fn add_block_to_ledger(
     block_index + 1
 }
 
-fn populate_ledger(blocks_config: &BlockConfig, ledger: &mut LedgerDB, watcher: &WatcherDB) {
+fn seed_block_provider(block_provider: &mut LocalBlockProvider<LedgerDB>) {
     let mut rng = thread_rng();
 
     let alice = AccountKey::random_with_fog(&mut rng);
     let recipients = vec![alice.default_subaddress()];
     // Origin block cannot have key images
-    add_block_to_ledger(ledger, &recipients, &[], &mut rng, watcher);
+    add_block_to_ledger(block_provider, &recipients, &[], &mut rng);
+}
+
+fn populate_block_provider(
+    block_provider: &mut LocalBlockProvider<LedgerDB>,
+    blocks_config: &BlockConfig,
+) {
+    let mut rng = thread_rng();
 
     for block in blocks_config {
         let recipients: Vec<_> = block.keys().cloned().collect();
         let key_images: Vec<_> = block.values().flat_map(|x| x.clone()).collect();
 
-        add_block_to_ledger(ledger, &recipients, &key_images, &mut rng, watcher);
+        add_block_to_ledger(block_provider, &recipients, &key_images, &mut rng);
     }
+
+    // The stores are running on separate threads. We wait twice as long as
+    // their POLL_INTERVAL to ensure they've had time to process the new blocks
+    std::thread::sleep(POLL_INTERVAL * 2);
```

Comment:
I'm not a fan of waits in tests, but I think I was able to correctly associate the wait time with why.
However I'm up for alternatives if one can think of better ways, I think it can only reliably be done with an API on the stores themselves to kick them directly instead of from the run loop.

Reply:
Seems like the API is CBB - maybe worth flagging as `// CBB: if the sleeps here are causing issues at some point in the future, we could instead develop an API to trigger XYZ`